src/programs/console_test/console_test.cpp

```diff
@@ -305,6 +307,86 @@ void MyTestApp::TestClipIntoFourier()
+		temp_line.logp = global_random_number_generator.GetUniformRandom() * 10000;
+		temp_line.microscope_spherical_aberration_mm = global_random_number_generator.GetUniformRandom() * 2.7;
+		temp_line.microscope_voltage_kv = global_random_number_generator.GetUniformRandom() * 300;
+		temp_line.occupancy = global_random_number_generator.GetUniformRandom() * 100;
+		temp_line.original_image_filename = wxString::Format("This_is_an_original_filename_string_with_a_random_number_:_%f", global_random_number_generator.GetUniformRandom() * 100000);
+		temp_line.particle_group =  myroundint(global_random_number_generator.GetUniformRandom() * 10);
+		temp_line.phase_shift = global_random_number_generator.GetUniformRandom() * 3.14;
+		temp_line.phi = global_random_number_generator.GetUniformRandom() * 180;
+		temp_line.pixel_size = global_random_number_generator.GetUniformRandom() * 2;
+		temp_line.position_in_stack = counter + 1;
+		temp_line.pre_exposure = global_random_number_generator.GetUniformRandom() * 10;
+		temp_line.psi = global_random_number_generator.GetUniformRandom() * 180;
+		temp_line.reference_3d_filename = wxString::Format("This_is_a_reference_3d_filename_string_with_a_random_number_:_%f", global_random_number_generator.GetUniformRandom() * 100000);
+		temp_line.score = global_random_number_generator.GetUniformRandom() * 100;
+		temp_line.sigma = global_random_number_generator.GetUniformRandom() * 180;
+		temp_line.stack_filename = wxString::Format("This_is_a_stack_filename_string_with_a_random_number_:_%f", global_random_number_generator.GetUniformRandom() * 100000);
+		temp_line.theta = global_random_number_generator.GetUniformRandom() * 180;
+		temp_line.total_exposure = global_random_number_generator.GetUniformRandom() * 100;
+		temp_line.x_shift = global_random_number_generator.GetUniformRandom() * 50;
+		temp_line.y_shift = global_random_number_generator.GetUniformRandom() * 50;
+
+		test_parameters.all_parameters.Add(temp_line);
+
+	}
+
+	temp_directory = wxFileName::GetTempDir();
+
+	// write star and binary file..
+
+	test_parameters.WriteTocisTEMStarFile(temp_directory + "/star_file.star");
+	test_parameters.WriteTocisTEMBinaryFile(temp_directory + "/binary_file.cistem");
+
+	// read in binary file and write out star file..
+
+	test_parameters.ClearAll();
+	test_parameters.ReadFromcisTEMBinaryFile(temp_directory + "/binary_file.cistem");
+	test_parameters.WriteTocisTEMStarFile(temp_directory + "/star_file_converted_from_binary.star");
+
+	// read in star file and write to binary..
+
+	test_parameters.ClearAll();
+	test_parameters.ReadFromcisTEMStarFile(temp_directory + "/star_file.star");
+	test_parameters.WriteTocisTEMBinaryFile(temp_directory + "/binary_file_converted_from_star.cistem");
+
+	// Check the sizes are the same - this isn't a very thorough test, but better than nothing.
+	// It is at least an easy way to get files with all the parameters written out to quickly check by eye
+
```

Comment:
How about comparing the files byte by byte? Something like this? https://stackoverflow.com/a/39097160/11267786

Reply:
Yeh I did think about this, the files are not exactly identical though as they contain the time they written.  I could probably ignore that bit quite easily.